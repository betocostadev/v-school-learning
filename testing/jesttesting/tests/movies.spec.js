import movies from '../src/movies'

describe('Favorite Movies', () => {
  let myMovies = []
  beforeEach(() => {
    myMovies = [
      {
        title: 'Age of Ultron',
        rate: null
      },
      {
        title: 'Gladiator',
        rate: 5
      }
    ]
  })

  test('can add a movie', () => {
    movies.add(myMovies, 'Kung Fury')
    expect(myMovies).toMatchSnapshot()
  })

  // To test only once:
  // test.only('can add a movie', () => {
  //   movies.add(myMovies, 'Kung Fury')
  //   expect(myMovies).toMatchSnapshot()
  // })
  // You can also use test.skip() to skip a test

  test('rate a movie', () => {
    movies.rate(myMovies[0], 3)
    expect(myMovies).toMatchSnapshot()
  })
})