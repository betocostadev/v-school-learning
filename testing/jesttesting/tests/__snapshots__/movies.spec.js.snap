// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Favorite Movies can add a movie 1`] = `
[
  {
    "rate": null,
    "title": "Age of Ultron",
  },
  {
    "rate": 5,
    "title": "Gladiator",
  },
  {
    "rate": null,
    "title": "Kung Fury",
  },
]
`;

exports[`Favorite Movies rate a movie 1`] = `
[
  {
    "rate": 3,
    "title": "Age of Ultron",
  },
  {
    "rate": 5,
    "title": "Gladiator",
  },
]
`;
